### ISSUE TITLE
Multi-array build error

### ISSUE THREAD

**Issue by Ruyk** [triSYCL/triSYCL#36]
When trying to build Parallel STL with TriSYCL I get the following error:

```
/home/ruyman/Projects/ruyk-parallel-stl/include/experimental/algorithm:287:38:   required from ‘InputIt std::experimental::parallel::find(ExecutionPolicy&&, InputIt, InputIt, T) [with ExecutionPolicy = sycl::sycl_execution_policy<FindAlgorithm>&; InputIt = __gnu_cxx::__normal_iterator<float*, std::vector<float> >; T = float]’
/home/ruyman/Projects/ruyk-parallel-stl/tests/find.cpp:63:59:   required from here
/usr/include/boost/multi_array.hpp:477:30: error: no matching function for call to ‘sycl::impl::search_result::search_result()’
     std::uninitialized_fill_n(base_,allocated_elements_,T());
```

This seems to be a problem with the multi_array interface. Which version of multi-array is required for triSYCL? I am using 1.58.

Travis Parallel STL Job output:
https://travis-ci.org/Ruyk/SyclParallelSTL/jobs/221684886

**Comment by a-doumoulakis** [comment]
Hello,

I don't think the problem comes from boost multi-array but rather from the definition of the structure```sycl::impl::search_result``` defined in *include/sycl/algorithm/find.hpp* of the Parallel STL project.

Adding a default constructor to the structure definition (as in the [trisycl branch](https://github.com/keryell/SyclParallelSTL/tree/triSYCL "The triSYCL branch") in @keryell's fork) should resolve this issue.
``` C++
typedef struct search_result {
  bool result;
  int index;
  search_result() = default;
  search_result(bool r, int i) : result(r), index(i) {}
} search_result;
```

Regards

**Comment by keryell** [comment]
We had a lengthy discussion on the interpretation of the SYCL specification between ComputeCpp and triSYCL during Khronos face-to-face meeting on SYCL in Amsterdam this week.
https://cvs.khronos.org/bugzilla/show_bug.cgi?id=16145

The conclusion was that for performance reasons, uninitialized buffer should just provide uninitialized memory and thus the default constructor should not be called.

So we need to change slightly the triSYCL behaviour...

**Comment by a-doumoulakis** [comment]
This issue is solved with https://github.com/triSYCL/triSYCL/commit/a2679b86e31a8691cdd97af889db090d05b93886 that also solved #38
